File: src/lecture1/lecture1_topic_py/scripts/subscriber.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import Int32

def callback(msg):
    rospy.loginfo("%d", msg.data)

def subscriber():
    rospy.init_node("subscriber", anonymous=True)
    rospy.Subscriber("topic", Int32, callback)
    rospy.spin()

if __name__ == '__main__':
    subscriber()

File: src/lecture1/lecture1_topic_cpp/src/subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

void callback(const std_msgs::Int32::ConstPtr& msg)
{
    ROS_INFO_STREAM(msg->data);
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "subscriber", ros::init_options::AnonymousName);
    ros::NodeHandle n;
    ros::Subscriber s = n.subscribe("topic", 5, callback);
    ros::spin();
    return 0;
}

